src/peft/tuners/reft/config.py

```diff
@@ -0,0 +1,100 @@
+            The layer indices to transform. If a list of ints is passed, it will apply the adapter to the layer indices
+            that are specified in this list. If a single integer is passed, it will apply the transformations on the
+            layer at this index.
+        layers_pattern (`str`):
+            The layer pattern name, used only if `layers_to_transform` is different from `None`.
+        rank_pattern (`dict`):
+            The mapping from layer names or regexp expression to ranks which are different from the default rank
+            specified by `r`.
+        modules_to_save (`List[str]`):
+            List of modules apart from adapter layers to be set as trainable and saved in the final checkpoint.
+        block_share (`bool`):
+            Whether to share the LoReFT parameters between blocks or not. This is `False` by default.
+    """
+    r: int = field(default=8, metadata={"help": "LoReFT rank"})
+    alpha: int = field(default=8, metadata={"help": "LoReFT alpha"})
+    loc: Optional[str] = field(default=None, metadata={"help": "token locations are applied LoReFT. e.g., f2+l2"})
+    dropout: float = field(
+        default=0.0, metadata={"help": "The dropout probability for disabling LoReFT modules during training"}
+    )
+    target_modules: Optional[Union[List[str], str]] = field(
+        default=None,
+        metadata={
+            "help": "List of module names or regex expression of the module names to replace with LoReFT."
+            "For example, ['q', 'v'] or '.*decoder.*(SelfAttention|EncDecAttention).*(q|v)$' "
+            "This can also be a wildcard 'all-linear' which matches all linear/Conv1D layers except the output layer."
+        },
+    )
+    layers_to_transform: Optional[Union[List[int], int]] = field(
+        default=None,
+        metadata={
+            "help": "The layer indexes to transform, is this argument is specified, PEFT will transform only the layers indexes that are specified inside this list. If a single integer is passed, PEFT will transform only the layer at this index."
+        },
+    )
+    layers_pattern: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "The layer pattern name, used only if `layers_to_transform` is different to None and if the layer pattern is not in the common layers pattern."
+        },
+    )
+    modules_to_save: Optional[List[str]] = field(
+        default=None,
+        metadata={
+            "help": "List of modules apart from ReFT layers to be set as trainable and saved in the final checkpoint. "
+            "For example, in Sequence Classification or Token Classification tasks, "
+            "the final layer `classifier/score` are randomly initialized and as such need to be trainable and saved."
+        },
+    )
+
+    block_share: bool = field(
+        default=False,
+        metadata={"help": "Whether to share the ReFT parameters between blocks or not."},
+    )
+
```

Comment:
So right now, LoReFT has no `init_weights` argument, unlike almost all other adapters. I think it would be useful to have this. I think you mentioned somewhere that by default, LoReFT is _not_ initialized to be the identity transform. In that case, we could set the default value to `False` (which in PEFT means _not_ identity), but it would still be nice for users to have the option to initialize as identity by passing `init_weights=True`.